tests/std/tests/P0355R7_calendars_and_time_zones_io/test.cpp

```diff
@@ -1247,8 +1247,52 @@ void test_io_manipulator() {
     fail_parse(WIDEN(CharT, "a  b"), CStringOrStdString{WIDEN(CharT, "a%nb")}, time);
 }
 
+namespace lwg_3956 {
+    struct has_adl_from_stream {
+        int value = 0;
+
+        template <class CharT, class Traits, class... ArgTypes>
+        friend basic_istream<CharT, Traits>& from_stream(

```

Comment:
Should include `<istream>` - perhaps surprisingly, `<sstream>` doesn't guarantee that it drags in `basic_istream`.